app/src/main/java/org/thoughtcrime/securesms/scribbles/UriGlideRenderer.java

```diff
@@ -0,0 +1,340 @@
+import org.signal.imageeditor.core.model.EditorElement;
+import org.signal.imageeditor.core.model.EditorModel;
+import org.thoughtcrime.securesms.mms.DecryptableStreamUriLoader;
+import org.thoughtcrime.securesms.mms.GlideApp;
+import org.thoughtcrime.securesms.mms.GlideRequest;
+import org.thoughtcrime.securesms.util.BitmapUtil;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Uses Glide to load an image and implements a {@link Renderer}.
+ *
+ * The image can be encrypted.
+ */
+public final class UriGlideRenderer implements SelectableRenderer {
+
+  private static final String TAG = Log.tag(UriGlideRenderer.class);
+
+  private static final int PREVIEW_DIMENSION_LIMIT = 2048;
+  private static final int MAX_BLUR_DIMENSION      = 300;
+
+  public static final float WEAK_BLUR   = 3f;
+  public static final float STRONG_BLUR = 25f;
+
+  private final Uri                     imageUri;
+  private final Paint                   paint                 = new Paint();
+  private final Matrix                  imageProjectionMatrix = new Matrix();
+  private final Matrix                  temp                  = new Matrix();
+  private final Matrix                  blurScaleMatrix       = new Matrix();
+  private final boolean                 decryptable;
+  private final int                     maxWidth;
+  private final int                     maxHeight;
+  private final float                   blurRadius;
+  private final RequestListener<Bitmap> bitmapRequestListener;
+
+  private boolean selected;
+
+  @Nullable private Bitmap bitmap;
+  @Nullable private Bitmap blurredBitmap;
+  @Nullable private Paint  blurPaint;
+
+  public UriGlideRenderer(@NonNull Uri imageUri, boolean decryptable, int maxWidth, int maxHeight) {
+    this(imageUri, decryptable, maxWidth, maxHeight, STRONG_BLUR);
+  }
+
+  public UriGlideRenderer(@NonNull Uri imageUri, boolean decryptable, int maxWidth, int maxHeight, float blurRadius) {
+    this(imageUri, decryptable, maxWidth, maxHeight, blurRadius, null);
+  }
+
+  public UriGlideRenderer(@NonNull Uri imageUri, boolean decryptable, int maxWidth, int maxHeight, float blurRadius, @Nullable RequestListener<Bitmap> bitmapRequestListener) {
+    this.imageUri              = imageUri;
+    this.decryptable           = decryptable;
+    this.maxWidth              = maxWidth;
+    this.maxHeight             = maxHeight;
+    this.blurRadius            = blurRadius;
+    this.bitmapRequestListener = bitmapRequestListener;
+    paint.setAntiAlias(true);
+    paint.setFilterBitmap(true);
+    paint.setDither(true);
+  }
+
+  @Override
+  public void render(@NonNull RendererContext rendererContext) {
+    if (getBitmap() == null) {
+      if (rendererContext.isBlockingLoad()) {
+        try {
+          Bitmap bitmap = getBitmapGlideRequest(rendererContext.context, false).submit().get();
+          setBitmap(rendererContext, bitmap);
+        } catch (ExecutionException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)